plugins/qc-s5gen2/fu-qc-s5gen2-hid-device.c

```diff
@@ -210,6 +210,51 @@ fu_qc_s5gen2_hid_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_qc_s5gen2_hid_device_setup(FuDevice *device, GError **error)
+{
+	guint idx;
+	guint16 vid = fu_device_get_vid(device);
+	guint16 pid = fu_device_get_pid(device);
+
+	/* FuHidDevice->setup */
+	if (!FU_DEVICE_CLASS(fu_qc_s5gen2_hid_device_parent_class)->setup(device, error))
+		return FALSE;
+
+	fu_device_add_instance_u16(device, "VID", vid);
+	fu_device_add_instance_u16(device, "PID", pid);
+
+	idx = fu_usb_device_get_manufacturer_index(FU_USB_DEVICE(device));
+	if (idx != 0x00) {
+		g_autofree gchar *tmp = NULL;
+		tmp = fu_usb_device_get_string_descriptor(FU_USB_DEVICE(device), idx, NULL);
```

Comment:
i think you need a `if (tmp != NULL)` here, no?

Reply:
No, if NULL be here the device wouldn't generate the GUID, meaning vendor did something wrong and we shouldn't update the device anyway.

Reply:
Ohh, in that case fu_device_build_instance_id_full() will fail as MANUFACTURER isn't set -- if we can avoid setting NULL then future us might want to set it in the quirk file or in a superclass or something.

Reply:
Exactly, the idea is -- if the device has BOTH manufacturer and product -- let's create the instance ID, otherwise -- please don't.
I'm really worrying to have intersections between different vendors, so let's set only known things.

But as you wish. If manufacturer is absent -- should I construct instance ID without it? The same question is about the product.